### DISCUSSION TITLE
Seasonality filter

### DISCUSSION THREAD

**Discussion by ezwelty** [falling-fruit/falling-fruit-web#932] [Ideas]
A seasonality filter is our most popular feature request. Although doing this properly – for all locations on the planet and all species in the database – will be an enormous undertaking, we could start with something much simpler (but less powerful). 

A few locations have `season_start` and `season_stop` (months) and we could begin by filtering on this information. Although this would hide most of the locations, it would be a welcome feature for the users taking the time to enter this information. There are also `observation.fruiting_status` associated with each location which we could use with a little more effort.

**Comment by tjbutler003** [comment]
Does anyone know of a database, I am thinking from the USDA or other governmental agency, that provides information on expected fruiting seasons?  Or maybe some sort of farmer's market based dataset?  I guess that might not capture all of the species available, but I can continue to look into this.

**Comment by ezwelty** [comment]
@tjbutler003 Thanks for asking. There are a lot of harvest calendars out there – you're right on with USDA and farmer's markets. Part of the challenge is mapping them to our type taxonomy and georeferencing them to a polygon or point and elevation so that we can write software that can predict the harvesting dates of the locations on the map.

Here is a list of calendars that we've found, and a draft template of how we may want to structure the data:
https://docs.google.com/spreadsheets/d/1Kbq0iBMRyjTEd0pCoCntQlKtdDDmC5stNhHLJV8xwYk/

**Comment by lordkiwi** [comment]
Might I suggest a strategy.  Each plant at the Longwood Gardens In Pa has
been tracked with over several years for blooms present and fruit present.
I suggest sampling allowing users to flag each observation with, blooms
present and with fruit present and fruit ripe.   You can then pool
observations based on USDA zones, sunset Zones , elevation or zip code.
After a few years the data will let everyone know area X ripens 2 weeks
before Area Y

https://plantexplorer.longwoodgardens.org//weboi/oecgi2.exe/INET_ECM_DispPl?NAMENUM=30569&DETAIL=1&startpage=1
https://plantexplorer.longwoodgardens.org//weboi/oecgi2.exe/INET_ECM_DispPl?NAMENUM=35084&DETAIL=1&startpage=1
https://plantexplorer.longwoodgardens.org//weboi/oecgi2.exe/INET_ECM_DispPl?NAMENUM=12298&DETAIL=1&startpage=1

https://plantexplorer.longwoodgardens.org/ecmweb/FindPlant.html

**Comment by ezwelty** [comment]
@lordkiwi Thanks for jumping in.

As you can see on our "add a review" form, we already collect fruiting status (Flowers, Unripe fruit, Ripe fruit):
https://fallingfruit.org/observations/new?location_id=2812

However, since 2013-12-13, when we launched that feature, we've only received 6,501 such observations (and only 2,915 for ripe fruit) – a great start, but not enough to cover all of the world's edible plant species in all possible locations on earth. I suspect a combined strategy will be needed to achieve full coverage.

I also think that a 2-week resolution is too optimistic. The standard deviation of "Ripe fruit" dates in spatially-clustered observations for the same species can be as high as 30 days – whether from urban microclimates or variability between individuals (different apple cultivars, for example), neither of which we can realistically resolve. So far, I've found that the best predictor of ripening day is latitude, followed weakly by temperature, while elevation has no predictive power whatsoever. I've tried temperature as both 1980-2010 mean, preceding winter mean, and preceding year mean, with similarly poor results – since I'm using a global 2.5 deg grid (NOAA NCEP), I suspect the resolution is simply not adequate and looking at using station data from NOAA GSOD instead.

There are also some issues with the flower-fruit sequence (e.g. desirable plant parts are not limited to flowers or fruit), so we're discussing how to revamp our observation fields here: https://github.com/falling-fruit/falling-fruit/issues/34

**Comment by lordkiwi** [comment]
I see the complexity your facing even attempting to make a prediction, and
revamped observation fields would be in order as asparagus and fiddlehead
ferns dont quite fit the fruit category.  At this time I suggest you do not
even work on a prediction model but focus on increasing the use of
observations.  For starters. with in the apps and the website begin asking
for observations. But not with a generic report observations it helps us
improve the site message. Even though its true you can be more subtle and
interesting.  When someone searches for mulberry trees ask them to report
there observations.  When using the map feature to search blink some of the
results and request observations, especially around the time that its
generally known. eg. Raspberries in may, June and July depending on climate
zone.  One way to build the observation culture might be tracking the
annual cherry blossoms, Azaleas and Dogwoods. While it is a tangent from
the fruit and edible mission it servers to build more participation in the
user base.  Another tangent would be fall interest, there are some blocks
in DC lined with Gingko tree but the window to know when the leaves have
turned is short.

Once that data set starts growing then spend time on predictions.


https://www.flickr.com/photos/streetsofdc/6338558934/

**Comment by tjbutler003** [comment]
I think the request for observation makes sense, and could be combined with the predictive model or even the most simple model of using current observations and like lordwiki mentions, sending observation requests maybe at the earliest standard deviation of ripening date, and maybe we can watch people mark different trees of the same species as ripening over time.  Might be ambitious as far as participation is concerned though.

**Comment by ezwelty** [comment]
I think significant progress could be made on this front by leveraging existing plant [phenology](https://en.wikipedia.org/wiki/Phenology) databases.

For example, see https://www.plantphenology.org, which brings together several datasources, including the [USA National Phenology Network](https://www.usanpn.org).

Edit: A simple interface by which Falling Fruit users could submit harvest months for different species for their region would be an efficient means of pre-populating this information coarsely over a wide range of geographies.

**Comment by mpaluta** [comment]
If Fruiting Status labels to train a predictive model is one solution, is it possible to gamify user contributions to incentivize making these observations? For example having local leaderboards of the highest contributors.

**Comment by ezwelty** [comment]
@mpaluta I think gamification could definitely help tease out more contributions, but I'm still not sure whether observations via Falling Fruit alone would be enough.

A Develop for Good team made progress towards this goal in summer 2023. They built predictive fruit ripeness models from historic climate data and phenology observations pulled from various sources: [dfg-seasons](https://github.com/falling-fruit/dfg-seasons). Now we need to find time to dig through the work and see if some of it can be made operational.

The current best description of the plan for a comprehensive seasonality filter is [here](https://docs.google.com/document/d/1ah6jKO9uizBqeBtTVoIXi51gpYEQEyYzzastcMKOp5Y/edit#heading=h.1hnrl61rsxes).
